src/main/java/com/thealgorithms/sorts/BeadSort.java

```diff
@@ -1,40 +1,53 @@
 package com.thealgorithms.sorts;
 
-// BeadSort Algorithm(wikipedia) : https://en.wikipedia.org/wiki/Bead_sort
-// BeadSort can't sort negative number, Character, String. It can sort positive number only
+import java.util.Arrays;
 
+/**
+ * BeadSort cannot sort negative numbers, characters, or strings.
+ * It only works for non-negative integers.
+ *
+ * @see <a href="https://en.wikipedia.org/wiki/Bead_sort">BeadSort Algorithm (Wikipedia)</a>
+ */
 public class BeadSort {
```

Comment:
The entire class could look something like that:
```java
public class BeadSort {
    /**
     * Sorts the given array using the BeadSort algorithm.
     *
     * @param array The array of non-negative integers to be sorted.
     * @return The sorted array.
     * @throws IllegalArgumentException If the array contains negative numbers.
     */
    public int[] sort(int[] array) {
        allInputsMustBeNonnegative(array);
        return extractSortedFromGrid(fillGrid(array));
    }

    private void allInputsMustBeNonnegative(final int[] array) {
        if (Arrays.stream(array).anyMatch(s -> s < 0)) {
            throw new IllegalArgumentException("BeadSort cannot sort negative numbers.");
        }
    }

    private char[][] fillGrid(final int[] array) {
        final var maxValue = Arrays.stream(array).max().orElse(0);
        var grid = getEmptyGrid(array.length, maxValue);

        int[] count = new int[maxValue];
        for (int i = 0, arrayLength = array.length; i < arrayLength; i++) {
            int k = 0;
            for (int j = 0; j < array[i]; j++) {
                grid[count[maxValue - k - 1]++][k] = '*';
                k++;
            }
        }
        return grid;
    }

    private char[][] getEmptyGrid(final int arrayLength, final int maxValue) {
        char[][] grid = new char[arrayLength][maxValue];
        for (int i = 0; i < arrayLength; i++) {
            for (int j = 0; j < maxValue; j++) {
                grid[i][j] = '-';
            }
        }

        return grid;
    }

    private int[] extractSortedFromGrid(final char[][] grid) {
        int[] sorted = new int[grid.length];
        for (int i = 0; i < grid.length; i++) {
            int k = 0;
            for (int j = 0; j < grid[grid.length - 1 - i].length && grid[grid.length - 1 - i][j] == '*'; j++) {
                k++;
            }
            sorted[i] = k;
        }
        return sorted;
    }
}
```

Please consider the datatype of `grid`: maybe it should be an array of `boolean` or some enum (enum should be more readable).

Reply:
Thanks. Fixed this using enum. It really more readable.